internal/sherlock/sherlock_integration_test.go

```diff
@@ -174,6 +179,201 @@ func Test_sherlockServerIntegration(t *testing.T) {
 			t.Errorf("Expected to receive code %d for non-existent service, got %d", http.StatusNotFound, response.Code)
 		}
 	})
+
+	t.Run("GET /builds", func(t *testing.T) {
+		defer func() {
+			if err := tools.Truncate(app.DB); err != nil {
+				t.Errorf("error truncatingdb in test run : %v", err)
+			}
+		}()
+
+		_, err := tools.SeedServices(app.DB)
+		if err != nil {
+			t.Fatalf("error seeding services: %v", err)
+		}
+
+		expectedBuilds, err := tools.SeedBuilds(app.DB)
+		if err != nil {
+			t.Fatalf("error seeding builds: %v", err)
+		}
+
+		expectedBuildsResponse := &builds.Response{Builds: expectedBuilds}
+
+		req, err := http.NewRequest(http.MethodGet, "/builds", nil)
+		if err != nil {
+			t.Errorf("error generating test GET /builds request: %v", err)
+		}
+
+		response := httptest.NewRecorder()
+
+		app.ServeHTTP(response, req)
+
+		assert.Equal(t, http.StatusOK, response.Code)
+
+		result := &builds.Response{}
+		if err := json.NewDecoder(response.Body).Decode(result); err != nil {
+			t.Errorf("error decoding response body: %v", err)
+		}
+
+		if diff := cmp.Diff(expectedBuildsResponse, result); diff != "" {
+			t.Errorf("unexpected difference in response body:\n%v", diff)
+		}
+	})
+	t.Run("POST /builds with pre-existing service", func(t *testing.T) {
+		defer func() {
+			if err := tools.Truncate(app.DB); err != nil {
+				t.Errorf("error truncatingdb in test run : %v", err)
+			}
+		}()
+
+		_, err := tools.SeedServices(app.DB)
+		if err != nil {
+			t.Fatalf("error seeding services: %v", err)
+		}
+
+		// create build for service prexisting in sherlock's db
+		newBuildRequest := builds.CreateBuildRequest{
+			VersionString: "gcr.io/broad/cromwell:1.0.1",
+			CommitSha:     "as2l3k",
+			BuildURL:      "https://jenkins.job/23",
+			BuiltAt:       time.Now(),
+			ServiceName:   "cromwell",
+			ServiceRepo:   "github.com/broadinstitute/cromwell",
+		}
+
+		payload := new(bytes.Buffer)
+		if err := json.NewEncoder(payload).Encode(newBuildRequest); err != nil {
+			t.Errorf("error encoding post payload: %v", err)
+		}
+
+		req, err := http.NewRequest(http.MethodPost, "/builds", payload)
+		if err != nil {
+			t.Fatalf("error constructing POST /builds request: %v", err)
+		}
+
+		response := httptest.NewRecorder()
+
+		app.ServeHTTP(response, req)
+
+		assert.Equal(t, http.StatusCreated, response.Code)
+		var result builds.Response
+		if err := json.NewDecoder(response.Body).Decode(&result); err != nil {
+			t.Errorf("error decoding response body: %v", err)
+		}
+
+		assert.Equal(t, 1, len(result.Builds))
+		assert.Empty(t, result.Error)
+		assert.Equal(t, newBuildRequest.VersionString, result.Builds[0].VersionString)
+	})
+	t.Run("POST /builds with new service", func(t *testing.T) {
```

Comment:
Love all this integration testing!

Can we add some coverage for a version conflict, i.e. client tries to publish a new build with a version string that already exists? Curious how user-friendly the error message is

Reply:
yep yep, will do

Reply:
I think currently it bubbles up an error from the DB itself saying that a `unique` constraint was violated. Could definitely wrap that in something more user friendly

Reply:
All set https://github.com/broadinstitute/sherlock/pull/8/commits/d65ae979a2c2b235277b5154a263e787ed61a999